modules/coin/rest/handlers.go

```diff
@@ -63,17 +63,34 @@ func doQueryAccount(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := proofs.OutputProof(account, proof.BlockHeight()); err != nil {
+	if err := proofs.FoutputProof(w, account, proof.BlockHeight()); err != nil {
 		common.WriteError(w, err)
-		return
 	}
-	common.WriteSuccess(w, account)
+}
+
+func PrepareSendTx(si *SendInput) basecoin.Tx {
+	tx := coin.NewSendOneTx(*si.From, *si.To, si.Amount)
+	// fees are optional
+	if si.Fees != nil && !si.Fees.IsZero() {
+		tx = fee.NewFee(tx, *si.Fees, *si.From)
+	}
+	// only add the actual signer to the nonce
+	signers := []basecoin.Actor{*si.From}
+	tx = nonce.NewTx(si.Sequence, signers, tx)
+	tx = base.NewChainTx(commands.GetChainID(), 0, tx)
+
+	if si.Multi {
+		tx = auth.NewMulti(tx).Wrap()
+	} else {
+		tx = auth.NewSig(tx).Wrap()
+	}
+	return tx
 }
 
 func doSend(w http.ResponseWriter, r *http.Request) {
 	defer r.Body.Close()
 	si := new(SendInput)
-	if err := common.ParseAndValidateRequestJSON(r, si); err != nil {
```

Comment:
Okay, one question...

This `ParseAndValidateRequestJSON` does some auto-magic validation logic on the input.
Then below there is validate logic done manually.

I think we should pick one way or the other. Like use your validation checks, but then only `ParseRequestJSON`.